backends/tfhe-cuda-backend/cuda/include/pbs/pbs_utilities.h

```diff
@@ -252,26 +254,29 @@ struct pbs_buffer_128<InputTorus, PBS_TYPE::CLASSICAL> {
   uint64_t *trivial_indexes;
 
   PBS_VARIANT pbs_variant;
-  bool uses_noise_reduction;
+  PBS_MS_REDUCTION_T
+  noise_reduction_type;
   bool gpu_memory_allocated;
 
   pbs_buffer_128(cudaStream_t stream, uint32_t gpu_index,
                  uint32_t lwe_dimension, uint32_t glwe_dimension,
                  uint32_t polynomial_size, uint32_t level_count,
                  uint32_t input_lwe_ciphertext_count, PBS_VARIANT pbs_variant,
-                 bool allocate_gpu_memory, bool allocate_ms_array,
-                 uint64_t &size_tracker) {
+                 bool allocate_gpu_memory,
+                 PBS_MS_REDUCTION_T noise_reduction_type,
+                 uint64_t &size_tracker)
+      : noise_reduction_type(noise_reduction_type) {
     gpu_memory_allocated = allocate_gpu_memory;
     cuda_set_device(gpu_index);
     this->pbs_variant = pbs_variant;
-    this->uses_noise_reduction = allocate_ms_array;
-    if (allocate_ms_array) {
+
+    if (noise_reduction_type == PBS_MS_REDUCTION_T::DRIFT) {
       this->temp_lwe_array_in = (InputTorus *)cuda_malloc_async(
```

Comment:
Seems strange that this is not using cuda_malloc_with_size_tracking, it should, and conditionally allocate if allocate_gpu_memory is true.

Reply:
Yeah, I don't know why I changed to a regular cuda_malloc_async. Fixed.